### PR TITLE
added some error handling to receipt controllers

### PR DESCRIPTION
**Pull request by singonwan**

### CODE REVIEW

File: server/controllers/receipt-controller.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const createReceipt = async (req, res, next) => {
 
         return res.json(receipt);
     } catch (err) {
-        const error = new HttpError('server error', 500);
+        const error = new HttpError('Server Error', 500);
```
Comment (varunrampal) [inline server/controllers/receipt-controller.js]: change 'server error' => 'Internal server error'

Comment (singonwan) [reply server/controllers/receipt-controller.js]: ok

File: server/controllers/receipt-controller.js [hunk 2]
```diff
@@ -32,16 +32,27 @@ const createReceipt = async (req, res, next) => {
...
+        // check
+        const receipt = await Receipt.findById(req.params.id);
+        if (!receipt) {
+            const error = new HttpError('Receipt not found', 404);
+            return next(error);
+        }
+        if (receipt.user != req.userData.userId) {
+            const error = new HttpError('Not Authorized', 401);
+            return next(error);
+        }
```
Comment (varunrampal) [inline server/controllers/receipt-controller.js:44]: what are you getting in receipt  const is a receipt object  and in line: 41 you are comparing receipt object with userId.

File: server/controllers/receipt-controller.js [hunk 3]
```diff
@@ -32,16 +32,27 @@ const createReceipt = async (req, res, next) => {
...
-        const receipt = await Receipt.findOneAndUpdate(
+        const newReceipt = await Receipt.findOneAndUpdate(
             { _id: req.params.id },
             req.body,
             { new: true }
         );
-        await receipt.save();
-        return res.json(receipt);
+        await newReceipt.save();
+        return res.json(newReceipt);
```
Comment (varunrampal) [inline server/controllers/receipt-controller.js:53]: I didn't get this logic?

File: server/controllers/receipt-controller.js (hunk 2 above)
Comment (singonwan) [reply server/controllers/receipt-controller.js:44]: receipit.user contains the user object id, thats for checking if the receipt belongs to the user

File: server/controllers/receipt-controller.js (hunk 3 above)
Comment (singonwan) [reply server/controllers/receipt-controller.js:53]: findOneAndUpdate's first param finds a receipt that has the id of the receipt id we are looking for, second param is to change the fields that are passed in from the req.body, third param makes sure that the newly updated object is returned

Comment (varunrampal) [reply server/controllers/receipt-controller.js:53]: ok, it's good but I will try to improve it more, for now you go with it. good job

**Review by varunrampal — APPROVED** [review]
It's looking, you go with it

**Pull request merged**